### ISSUE TITLE
Redis Microservice: Concurrent messages are not handled correctly.

### ISSUE THREAD

**Issue by aleksey-taranets** [nestjs/nest#2988]
## Current behavior
Concurrent microservice messages are not handled correctly.
I see this behavior in "ClientRedis" class.
When we do several concurrent messages the handler creates "responseChannel". The "publish" handler subscribes on "responseChannel" and waits for a response. When it gets the response the handler `unsubscribe(responseChannel)`.
In this case for concurrent requests, we get only one response, other responses are lost

## Input Code

```ts
class ClientRedis {
// other methods of the class

publish(partialPacket, callback) {
        try {
            const packet = this.assignPacketId(partialPacket);
            const pattern = this.normalizePattern(partialPacket.pattern);
            const serializedPacket = this.serializer.serialize(packet);
            const responseChannel = this.getResPatternName(pattern);
            this.routingMap.set(packet.id, callback);
            this.subClient.subscribe(responseChannel, (err) => {
                if (err) {
                    return;
                }
                this.pubClient.publish(this.getAckPatternName(pattern), JSON.stringify(serializedPacket));
            });
            return () => {
                this.subClient.unsubscribe(responseChannel); // <- the problem is here, because of not unique channels
                this.routingMap.delete(packet.id);
            };
        }
        catch (err) {
            callback({ err });
        }
    }
}
```

## Expected behavior
All concurrent messages should be received

## Possible Solution
"responseChannel" is generated from as a string: `${pattern}_res`
There is a possible solution that we can generate unique channels for different messages with the same pattern

## Environment

<pre><code>
Nest version: 6.7.1

 
For Tooling issues:
- Node version: v10.16.0  
- Platform: Linux  

</code></pre>

**Comment by aleksey-taranets** [comment]
Posible hotfix:
concurrentСounter counts the same pattern and won't allow to unsubscribe the channel until all patterns have got a response
```ts
publish(partialPacket, callback) {
        try {
            const packet = this.assignPacketId(partialPacket);
            const pattern = this.normalizePattern(partialPacket.pattern);
            const serializedPacket = this.serializer.serialize(packet);
            const responseChannel = this.getResPatternName(pattern);
            this.routingMap.set(packet.id, callback);
            // concurrentСounter counts the same pattern and won't allow to unsubscribe the channel until all patterns have got a response
            let concurrentСounter = this.concurrentPatern.get(pattern) || 0;
            this.concurrentPatern.set(pattern, ++concurrentСounter);
            this.subClient.subscribe(responseChannel, (err) => {
                if (err) {
                    return;
                }
                this.pubClient.publish(this.getAckPatternName(pattern), JSON.stringify(serializedPacket));
            });
            return () => {
                let concurrentСounter = this.concurrentPatern.get(pattern);
                if (concurrentСounter === 1) {
                    this.concurrentPatern.delete(pattern);
                    this.subClient.unsubscribe(responseChannel);
                } else {
                    this.concurrentPatern.set(pattern, --concurrentСounter);
                }
                this.routingMap.delete(packet.id);
            };
        }
        catch (err) {
            callback({ err });
        }
    }
```

**Comment by kamilmysliwiec** [comment]
Thanks for reporting, fixed in 6.7.2 :)
